elements/lisk-bft/src/bft.ts

```diff
@@ -146,18 +143,27 @@ export class BFT extends EventEmitter {
 		return ForkStatus.DISCARD;
 	}
 
-	public isBFTProtocolCompliant(blockHeader: BlockHeader, stateStore: StateStore): boolean {
+	// eslint-disable-next-line class-methods-use-this
+	public async isBFTProtocolCompliant(
+		blockHeader: BlockHeader,
+		stateStore: StateStore,
+	): Promise<boolean> {
 		assert(blockHeader, 'No block was provided to be verified');
 
 		const roundsThreshold = 3;
-		const heightThreshold = this.constants.activeDelegates * roundsThreshold;
+		const validators = await getValidators(stateStore);
+		const numberOfVotingValidators = validators.filter(
+			validator => validator.isConsensusParticipant,
```

Comment:
I believe that's the property we discussed earlier related to boolean weight value.

Reply:
yes! with the discussion with research team, i updated